webapp/src/ts/services/create-user-for-contacts.service.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable } from '@angular/core';
+import { DbService } from '@mm-services/db.service';
+import { DBSyncService, SyncStatus } from '@mm-services/db-sync.service';
+import { SessionService } from '@mm-services/session.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { UserContactService } from '@mm-services/user-contact.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateUserForContactsService {
+  private currentUsername: string;
+
+  constructor(
+    private settingsService: SettingsService,
+    private userContactService: UserContactService,
+    private dbService: DbService,
+    private dbSyncService: DBSyncService,
+    private sessionService: SessionService,
+  ) {
+    this.settingsService
+      .get()
+      .then((settings) => {
+        if (settings?.transitions?.create_user_for_contacts) {
+          this.dbSyncService.subscribe(async (status) => {
+            return this.syncStatusChanged(status);
+          });
+        }
+      });
+  }
+
+  setReplaced(originalContact, newContact) {
+    if (!originalContact) {
+      throw new Error('The original contact could not be found when replacing the user.');
+    }
+    if (!newContact) {
+      throw new Error('The new contact could not be found when replacing the user.');
+    }
+    if (!originalContact.parent?._id || originalContact.parent._id !== newContact.parent._id) {
+      throw new Error('The new contact must have the same parent as the original contact when replacing a user.');
+    }
+    const originalUsername = this.getCurrentUsername();
+    if (!originalUsername) {
+      throw new Error('The current username could not be found when replacing the user.');
+    }
+
+    const status = this.sessionService.isOnlineOnly() ? UserCreationStatus.READY : UserCreationStatus.PENDING;
```

Comment:
Documentation says we only support replacing offline users. Should we just have this function do nothing when an attempt to replace an online user is made?

Reply:
Really the only reason we don't support replacing online users is because client-side transitions [currently only run for offline users](https://github.com/medic/cht-core/blob/4382617651cd92a1c244557f20f397301ccdfb48/webapp/src/ts/app.component.ts#L299).  By the time we realized that, we already had this code in place to support the online user flow.

It seems pretty likely, in the future that we have a request to support this feature for online users (at which point we will have to sort out any challenges with running (or not running) the other transitions for online users). It seemed fine to leave this bit of code (and its tests) in place since it will be useful when we do support online users.  

(Also worth noting that we should never be able to accidentally start supporting replacing online users since we have an e2e test that is intended to fail in this particular case.)